### ISSUE TITLE
Allow opening multiple files while keeping the cmd-P prompt open with right arrow key

### ISSUE THREAD

**Issue by wzxu** [zed-industries/zed#4864]
### Check for existing issues

- [X] Completed

### Describe the feature

Currently in the cmd-P prompt ("Search project files…") we can only press enter to open one file at a time.
 
In other editors like Sublime Text or VS Code, you can press right arrow key to open a file while keeping the prompt open, allowing you to keep opening multiple files with similar names. Zed should support this as well.

**Comment by LittleFat3Tuan** [comment]
It's been over a year. Please. We need this.

**Comment by daniele-orlando** [comment]
Hello everyone. Is there any progress on this issue?

**Comment by stevenchanin** [comment]
+1 for this. I frequently run into this situation. In VS code you can do this by hitting the right arrow (open file without closing file search) or command-right arrow (open file into other pane without closing file search). It's a big convenience.

**Comment by stevenchanin** [comment]
The PR seems to be just sitting there. If you'd like this issue fixed, could you go to the PR and give it a +1?

**Comment by init05** [comment]
Hey everyone. Looks like #38914 has been sitting for a while without getting across the finish line, which is a shame because this is one of those small quality-of-life things I genuinely miss from VS Code on a daily basis.

I went ahead and opened a fresh PR: #57258

It addresses the feedback from the previous one — specifically, it uses the right arrow key at the end of the input to trigger the action (just like VS Code and Sublime), which is now possible thanks to the end_of_input context that was added to the editor since that PR was opened. Each opened file also becomes the active tab so you get visual feedback every time, without the finder losing focus. There's also a "Keep Open" button in the footer for mouse users.

If this is still something people want, would love a review or a 👍 on the PR

**Comment by ackmanx** [comment]
Actually never heard of the right arrow to open with keeping the modal open in other editors before, so that's neat to learn.

Coming from Intellij, I was trying to open multiple files in Zed by holding shift and selecting multiple files from the results to press _Enter_ on to open. For example, if one wants to open up 12 files, it's easier to highlight all of them vs having to go down, right, down, right, down, right etc.

**Comment by daniele-orlando** [comment]
> Actually never heard of the right arrow to open with keeping the modal open in other editors before, so that's neat to learn.
> 
> Coming from Intellij, I was trying to open multiple files in Zed by holding shift and selecting multiple files from the results to press _Enter_ on to open. For example, if one wants to open up 12 files, it's easier to highlight all of them vs having to go down, right, down, right, down, right etc.

I agree, having both ways for opening multiple files would be great:
- Down+Shift for fast opening top 2/3 results
- Shift + Select for opening many files

**Comment by wzxu** [comment]
Hi @TomPlanche I think this did work for a while, but it no longer work for me in latest preview, can you reproduce?

Zed Preview 1.9.0 
Commit: 0a64bc9d6dfa30e429767f2f62d0eb12efb7defa
Version: 1.9.0+preview.313.0a64bc9d6dfa30e429767f2f62d0eb12efb7defa

**Comment by daniele-orlando** [comment]
@ChristopherBiscardi Please reopen this issue as this feature no longer works in latest **Zed 1.10.0**.

```
Version: 1.10.0+stable.318.111c4082fd38215c31fa87803ece7695f898a94e
```

cc: @TomPlanche @Dima-369

**Comment by stevenchanin** [comment]
I just tested on MacOS (26.5.2) and Zed (1.10.0) and it is NOT working. It definitely worked for a while. :-(
